logs_test.go

```diff
@@ -21,18 +21,43 @@ import (
 	"time"
 
 	"github.com/containerd/nerdctl/pkg/testutil"
+	"github.com/pkg/errors"
 )
 
 func TestLogs(t *testing.T) {
 	base := testutil.NewBase(t)
 	const containerName = "nerdctl-test-logs"
+	const containerName2 = "nerdctl-test-logs-2"
+	const expected = `foo
+bar`
+
 	defer base.Cmd("rm", containerName).Run()
 	base.Cmd("run", "-d", "--name", containerName, testutil.AlpineImage,
 		"sh", "-euxc", "echo foo; echo bar").AssertOK()
+
 	time.Sleep(3 * time.Second)
 	base.Cmd("logs", "-f", containerName).AssertOut("bar")
 	// Run logs twice, make sure that the logs are not removed
 	base.Cmd("logs", "-f", containerName).AssertOut("foo")
+
+	//test timestamps flag
+	base.Cmd("logs", "-t", containerName).AssertOut(time.Now().Format("2006-01-02"))
+
+	//test tail flag
+	base.Cmd("logs", "-n", "all", containerName).AssertOut(expected)
+	base.Cmd("logs", "-n", "1", containerName).AssertOutWithFunc(func(stdout string) error {
+		if !(stdout == "bar\n" || stdout == "") {
+			return errors.Errorf("expected %q or %q, got %q", "bar", "", stdout)
+		}
+		return nil
+	})
+
+	//test since / until flag
+	base.Cmd("logs", "--since", "1s", containerName).AssertOut("")
```

Comment:
AssertOut(“”) seems NOP, I guess we need AssertNoOut(expected) here.
(And we should rename AssertOut to AssertOutContains, eventually)

Reply:
fixed.